aten/src/ATen/core/ivalue.cpp

```diff
@@ -953,9 +953,31 @@ std::vector<std::reference_wrapper<const at::DataPtr>> ivalue::Future::extractDa
   if (value.isPyObject()) {
     std::vector<at::Tensor> tensors =
         value.toPyObjectHolder()->extractTensors();
-    data_ptrs.reserve(tensors.size());
+    auto num_data_ptrs = 0;
```

Comment:
The argument to `reserve` takes a `size_t` but by using `auto` this will be inferred as `int`, which is "probably" fine but sometimes it could be a smaller type than what we need and, moreover, it's signed whereas `size_t` is unsigned. I think using `size_t` explicitly would be safer.

In general I tend to dislike `auto` because while it's easier to write it makes the code much harder to read. (And, in general, code is read much more often than it's written, which is where the concept of "readability" comes from). I find that `auto` hides the real type of things and could thus sometimes allow an "unexpected" inferred type to slip through unnoticed.

Reply:
okay Ill update it.